app/controllers/billing_expenses_controller.rb

```diff
@@ -1,38 +1,41 @@
 class BillingExpensesController < ApplicationController
-  before_action :set_billing_expense, only: [:show, :edit, :update, :destroy]
+  before_action :set_billing_expense, only: [:show, :destroy]
   before_action :set_volunteer
+  before_action :set_volunteer_hours, only: :create
+  after_action :volunteer_hours_update, only: :create
 
   def index
     authorize BillingExpense
     @billing_expenses = BillingExpense.where(volunteer: @volunteer)
   end
 
-  def show; end
+  def show
+    respond_to do |format|
+      format.html
+      format.pdf do
+        render pdf:
+          "Spesenauszahlung-#{@volunteer.contact.first_name}-#{@volunteer.contact.last_name}-#{@volunteer.hours.last.meeting_date}",
+          template: 'billing_expenses/show.html.slim',
+          layout: 'pdf.pdf', encoding: 'UTF-8'
+      end
+    end
+  end
 
   def new
     @billing_expense = BillingExpense.new(volunteer: @volunteer)
     authorize @billing_expense
   end
 
-  def edit; end
-
   def create
     @billing_expense = BillingExpense
                        .new(billing_expense_params.merge(@volunteer.slice(:bank, :iban)))
     @billing_expense.user = current_user
+    @billing_expense.amount = compute_hours
     authorize @billing_expense
     if @billing_expense.save
       redirect_to @volunteer, make_notice
     else
-      render :new
-    end
-  end
-
-  def update
-    if @billing_expense.update(billing_expense_params)
-      redirect_to @volunteer, make_notice
-    else
-      render :edit
+      redirect_to @volunteer, notice: t('already_computed')
```

Comment:
I'm not sure: but would it make more sense to `redirect_to @billing_expense` instead?

Reply:
for me both is fine. maybe @chrysalag can comment on this, why she preferred to implement it this way :)

Reply:
Yes, it probably makes more sense to `redirect to @billing_expense`